src/main/java/seedu/address/ui/NewsCard.java

```diff
@@ -0,0 +1,74 @@
+package seedu.address.ui;
+
+import java.util.logging.Logger;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.events.ui.NewsCardClickedEvent;
+
+/**
+ * An UI component that displays information of a news link from www.cryptopanic.com.
+ */
+public class NewsCard extends UiPart<Region> {
+
+    private static final String FXML = "NewsListCard.fxml";
+
+    private final Logger logger = LogsCenter.getLogger(this.getClass());
+
+    /**
+     * Note: Certain keywords such as "location" and "resources" are reserved keywords in JavaFX.
+     * As a consequence, UI elements' variable names cannot be set to such keywords
+     * or an exception will be thrown by JavaFX during runtime.
+     *
+     * @see <a href="https://github.com/se-edu/addressbook-level4/issues/336">The issue on AddressBook level 4</a>
+     */
+
+    @FXML
+    private HBox cardNewsPane;
+    @FXML
+    private Label headline;
+    @FXML
+    private Label pubdate;
+    private String url;
+
+    public NewsCard(String title, String date, String url) {
+        super(FXML);
+        this.headline.setText(title);
+        this.pubdate.setText(date);
+        this.url = url;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof NewsCard)) {
+            return false;
+        }
+
+        // state check
+        NewsCard card = (NewsCard) other;
+        return this.headline.equals(card.headline)
+                && this.pubdate.equals(card.pubdate)
+                && this.url.equals(card.url);
+    }
+
+    /**
+     * Handles the mouse click event, {@code mouseEvent}.
+     */
+    @FXML
+    private void handleMouseClick(MouseEvent mouseEvent) {
+        if (mouseEvent.isStillSincePress()) {
```

Comment:
You'll want to trigger on an actual click event. cf. https://docs.oracle.com/javase/8/javafx/api/javafx/scene/input/MouseEvent.html#getEventType--
with the https://docs.oracle.com/javase/8/javafx/api/javafx/scene/input/MouseEvent.html#MOUSE_CLICKED event type.

Consider what happens if the user: 
- Clicks on a panel, then drags and releases on another (should not trigger)
- Clicks on another panel, then releases on this one (should not trigger)
- Clicks on a panel and loses focus (should not trigger)
- Clicks down, moves mouse around the screen and releases on the same panel (should trigger)
- etc..